src/terrain/tile.hpp

```diff
@@ -60,21 +60,20 @@ class Tile {
     uint16_t z; // The z index
     // does this need to know where it is?
     // The tile color is determined by this and the material type.
-    uint8_t color_id;
+    uint8_t color_id_;
     // Determined by the horizontal manhattan distance from a wall
-    uint8_t grow_data_high;
+    uint8_t grow_data_high_;
     // Determined by the horizontal manhattan distance from a edge
-    uint8_t grow_data_low;
+    uint8_t grow_data_low_;
 
-    bool grow_sink;   // not used
-    bool grow_source; // not used
-    bool grass;       // Does this tile obey grass color Gradient?
+    bool grow_sink_;   // not used
+    bool grow_source_; // not used
+    bool grass_;       // Does this tile obey grass color Gradient?
```

Comment:
You should use [bit fields](https://en.cppreference.com/w/cpp/language/bit_field)

Reply:
Ya probably optimized, but it can't hurt. The sizing may have to do with your struct ordering. You may want to try re-ordering fields (biggest -> smallest) with bit-fields. Bit-fields only work if the members are consecutive.